libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1681,352 +1681,556 @@ void TypeChecker::endVisit(BinaryOperation const& _operation)
+				msg +=
+					" This function requires a single bytes argument."
+					" If all your arguments are value types, you can use"
+					" abi.encode(...) to properly generate it.";
 		}
 		else if (
-			functionType->kind() == FunctionType::Kind::KECCAK256 ||
-			functionType->kind() == FunctionType::Kind::SHA256 ||
-			functionType->kind() == FunctionType::Kind::RIPEMD160
+			_functionType->kind() == FunctionType::Kind::KECCAK256 ||
+			_functionType->kind() == FunctionType::Kind::SHA256 ||
+			_functionType->kind() == FunctionType::Kind::RIPEMD160
 		)
 			msg +=
 				" This function requires a single bytes argument."
-				" Use abi.encodePacked(...) to obtain the pre-0.5.0 behaviour"
-				" or abi.encode(...) to use ABI encoding.";
+				" Use abi.encodePacked(...) to obtain the pre-0.5.0"
+				" behaviour or abi.encode(...) to use ABI encoding.";
 		m_errorReporter.typeError(_functionCall.location(), msg);
+		return;
 	}
-	else if (isPositionalCall)
-	{
-		for (size_t i = 0; i < arguments.size(); ++i)
-		{
-			auto const& argType = type(*arguments[i]);
-			if (functionType->takesArbitraryParameters() && i >= parameterTypes.size())
-			{
-				bool errored = false;
-				if (auto t = dynamic_cast<RationalNumberType const*>(argType.get()))
-					if (!t->mobileType())
-					{
-						m_errorReporter.typeError(arguments[i]->location(), "Invalid rational number (too large or division by zero).");
-						errored = true;
-					}
-				if (!errored && !argType->fullEncodingType(false, abiEncoderV2, !functionType->padArguments()))
-					m_errorReporter.typeError(arguments[i]->location(), "This type cannot be encoded.");
-			}
-			else if (!type(*arguments[i])->isImplicitlyConvertibleTo(*parameterTypes[i]))
-			{
-				string msg =
-					"Invalid type for argument in function call. "
-					"Invalid implicit conversion from " +
-					type(*arguments[i])->toString() +
-					" to " +
-					parameterTypes[i]->toString() +
-					" requested.";
-				if (
-					functionType->kind() == FunctionType::Kind::BareCall ||
-					functionType->kind() == FunctionType::Kind::BareCallCode ||
-					functionType->kind() == FunctionType::Kind::BareDelegateCall ||
-					functionType->kind() == FunctionType::Kind::BareStaticCall
-				)
-					msg += " This function requires a single bytes argument. If all your arguments are value types, you can use abi.encode(...) to properly generate it.";
-				else if (
-					functionType->kind() == FunctionType::Kind::KECCAK256 ||
-					functionType->kind() == FunctionType::Kind::SHA256 ||
-					functionType->kind() == FunctionType::Kind::RIPEMD160
-				)
-					msg +=
-						" This function requires a single bytes argument."
-						" Use abi.encodePacked(...) to obtain the pre-0.5.0 behaviour"
-						" or abi.encode(...) to use ABI encoding.";
-				m_errorReporter.typeError(arguments[i]->location(), msg);
-			}
-		}
-	}
+
+	// Parameter to argument map
+	size_t const param_arg_map_size = parameterTypes.size();
```

Comment:
Please use camelCase, or someone could also argue that this variable is not necessary at all (don't mind the complexity).

Reply:
It used to be a plain C array (i.e. no size()) hence the size variable.
Removed the size var and camel cased param_arg_map.